ultravox/data/datasets.py

```diff
@@ -492,78 +505,95 @@ def _get_sample(self, row: transformers.BatchFeature) -> VoiceSample:
+        query_prompt = self._choice(self.QUERY_PREFIX)
+        question = "<|audio|>" if self._args.include_audio else question_str
+        prompt = f"{query_prompt}{question}{separator}{prompt}"
 
         if self._args.include_context:
-            context_prompt = self._choice(self.CONTEXT_PROMPTS)
-            prompt = f"{context_prompt}{{context}}{separator}{prompt}"
+            context_prompt = self._choice(self.CONTEXT_PREFIX)
+            prompt = f"{context_prompt}{context}{separator}{prompt}"
 
-        return prompt
+        return prompt.strip()
+
+
+class BoolQWithExtendedAnswerDataset(BoolQDataset, QAVoiceDatasetMixin):
+    """
+    A version of BoolQ that includes the context in the prompt and a longer explanation in the answer.
+    """
+
+    PROMPT_SUFFIXES = [
+        "Provide a short explanation, then respond with True/False on the last line",
+        "Explain briefly, concluding with True/False on a new line."
+        "Write a quick explanation, and finish with True/False on the last line"
+        "Summarize in a few words, and end with True/False on a new line."
+        "Give a brief explanation first, then answer with True/False on the final line",
+        "Start with a concise explanation, and end with a True/False response on the last line.",
+        "Explain briefly and follow up with True/False at the end",
+        "Write a short explanation, then state True/False on a new line.",
+        "First, offer a brief explanation, and then reply with True/False at the end.",
+        "Present a concise explanation, ending with a True/False answer on the final line",
+        "Start with a brief explanation, and then answer with True/False at the end.",
+    ]
+
+    def _get_sample(self, row: transformers.BatchFeature) -> Optional[VoiceSample]:
+        """
+        Example conversation:
+            <|user|> Passage: {context}
+            Question: {question}
+            Provide a short explanation, then respond with True/False on the last line
+            <|assistant|> {short_explanation}
+            Answer: {answer}
+        """
+        if len(row["passage"]) > self._args.max_context_length:
+            # Skip samples with long context
+            return None
 
-    def _get_sample(self, row: transformers.BatchFeature) -> VoiceSample:
         answer = "True" if row["answer"] else "False"
-        answer_prompt = self._choice(self.ANSWER_PROMPTS)
-        query_prompt = self._get_query_prompt()
-        user_content = query_prompt.format(
-            question="<|audio|>" if self._args.include_audio else row["question"],
-            context=row["passage"],
+        answer_prompt = self._choice(self.ANSWER_PREFIX)
+        user_message = self._get_query_prompt(
+            question_str=row["question"], context=row["passage"]
         )
         messages = [
-            {"role": "user", "content": user_content},
+            {"role": "user", "content": user_message},
```

Comment:
suggest adding a builder method like `_get_messages(user, assistant)` so you don't need to build this message array by hand in each class.